dolphin-dnn/src/test/java/edu/snu/dolphin/dnn/layers/PoolingLayerTest.java

```diff
@@ -0,0 +1,159 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package edu.snu.dolphin.dnn.layers;
+
+import edu.snu.dolphin.dnn.blas.Matrix;
+import edu.snu.dolphin.dnn.blas.MatrixFactory;
+import edu.snu.dolphin.dnn.blas.jblas.MatrixJBLASFactory;
+import edu.snu.dolphin.dnn.conf.*;
+import org.apache.reef.tang.Configuration;
+import org.apache.reef.tang.Tang;
+import org.apache.reef.tang.exceptions.InjectionException;
+import org.junit.Before;
+import org.junit.Test;
+import edu.snu.dolphin.dnn.conf.LayerConfigurationParameters.*;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Test class for pooling layer.
+ */
+public final class PoolingLayerTest {
+
+  private static MatrixFactory matrixFactory;
+  private static final float TOLERANCE = 1e-6f;
+
+  static {
+    final Configuration configuration = Tang.Factory.getTang().newConfigurationBuilder()
+                .bindImplementation(MatrixFactory.class, MatrixJBLASFactory.class)
+                .build();
+    try {
+      matrixFactory = Tang.Factory.getTang().newInjector(configuration).getInstance(MatrixFactory.class);
+    } catch (final InjectionException e) {
+      throw new RuntimeException("InjectionException while injecting a matrix factory: " + e);
+    }
+  }
+
+  private final Matrix input = matrixFactory.create(new float[][]{
+      {0, 10},
+      {9, 22},
+      {2, 13},
+      {8, 0},
+      {3, 5},
+      {7, 3},
+      {10, 6},
+      {4, 7},
+      {1, 9}});
+  private final Matrix expectedMaxPoolingActivation = matrixFactory.create(new float[][]{
+      {9, 22},
+      {9, 22},
+      {10, 7},
+      {7, 9}});
+  private final Matrix expectedAveragePoolingActivation = matrixFactory.create(new float[][]{
+      {5, 9.25f},
+      {5.25f, 10.75f},
+      {6.25f, 4.5f},
+      {3.75f, 6}});
+  private final Matrix nextError = matrixFactory.create(new float[][]{
+      {12, 0},
+      {16, 4},
+      {20, 12},
+      {4, 8}});
+  private final Matrix expectedMaxPoolingError = matrixFactory.create(new float[][]{
+      {0, 0},
+      {28, 4},
+      {0, 0},
+      {0, 0},
+      {0, 0},
+      {4, 0},
+      {20, 0},
+      {0, 12},
+      {0, 8}});
+  private final Matrix expectedAveragePoolingError = matrixFactory.create(new float[][]{
+      {3, 0},
+      {7, 1},
+      {4, 1},
+      {8, 3},
+      {13, 6},
+      {5, 3},
+      {5, 3},
+      {6, 5},
+      {1, 2}});
+
+  private LayerBase maxPoolingLayer;
+  private LayerBase averagePoolingLayer;
+
+  @Before
+  public void setup() throws InjectionException {
+    final Configuration layerConfMax = Tang.Factory.getTang().newConfigurationBuilder()
+        .bindNamedParameter(LayerIndex.class, String.valueOf(0))
+        .bindNamedParameter(LayerInputShape.class, "3,3")
+        .bindImplementation(MatrixFactory.class, MatrixJBLASFactory.class)
+        .build();
+
+    final Configuration layerConfAverage = Tang.Factory.getTang().newConfigurationBuilder()
```

Comment:
If `layerConfMax` and `layerConfAverage` are same, why don't we have a single layer configuration  `layerConf` and use it for injecting both max pooling layer and average pooling layer?